src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm

```diff
@@ -137,12 +176,41 @@ CHIP_ERROR OnMessageToSend(TransferSession::OutputEvent & event)
         }
 
         auto & msgTypeData = event.msgTypeData;
-        return mExchangeCtx->SendMessage(msgTypeData.ProtocolId, msgTypeData.MessageType, std::move(event.MsgData), sendFlags);
+        // If there's an error sending the message, close the exchange and call ResetState.
+        // TODO: If we can remove the !mInitialized check in ResetState(), just calling ResetState() will suffice here.
+        CHIP_ERROR err
+            = mExchangeCtx->SendMessage(msgTypeData.ProtocolId, msgTypeData.MessageType, std::move(event.MsgData), sendFlags);
+        if (err != CHIP_NO_ERROR) {
+            mExchangeCtx->Close();
+            mExchangeCtx = nullptr;
+            ResetState();
+        }
+        // If we sent a status report and it was successful, set the Exchange context to null and call ResetState.
```

Comment:
This comment in large part re-state the code. How about: `if the send was successful, ExchangeContext is nulled to prevent the exchange from being closed by this object during ResetState, as we need the message to be transmitted`.

Having said that though: if Close is not called by this object, what calls closed? That should probably part of the comment.

Reply:
The exchange closes itself on send-without-reply-expected.  That's part of the exchange contract.

Reply:
added comment here